src/gtk/mimetype.cpp

```diff
@@ -0,0 +1,54 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        src/gtk/mimetype.cpp
+// Purpose:     classes and functions to manage MIME types
+// Author:      Hans Mackowiak
+// Modified by:
+// Created:     05.06.2016
+// Copyright:   (c) 2016 Hans Mackowiak <hanmac@gmx.de>
+// Licence:     wxWindows licence (part of wxExtra library)
+/////////////////////////////////////////////////////////////////////////////
+
+
+#include "wx/wxprec.h"
+
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#if wxUSE_MIMETYPE && wxUSE_FILE
+
+#include "wx/gtk/mimetype.h"
+
+#include <gio/gio.h>
+#include <gtk/gtk.h>
+
+#include "wx/gtk/private/string.h"
+
+wxString wxGTKMimeTypesManagerImpl::GetIconFromMimeType(const wxString& mime)
+{
+    wxString icon;
+    wxGtkString type(g_content_type_from_mime_type(mime.mb_str()));
+
+    GIcon *gicon(g_content_type_get_icon(type));
```

Comment:
Using `wxGtkObject` from `include/wx/gtk/private/object.h` would be better than explicit `g_object_unref()` (below as well).